### PR TITLE
wip - feat: update global scrollbar styles, add dark mode COMPASS-5597

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-5597
This pr:
- Adds a new hook `useScrollbars` that passes a `className` with the current theme's scrollbar styles in it. We then use this hook inside of compass-home to style the scrollbars based on the theme. Places with darkmode, like aggregations use this inside of their own theme provider so it gets the new colors.
- Removes most scrollbar overrides (for example on preview documents and the performance page).
- Updates the scrollbar styles so a vertical scrollbar appears when the content is hovered (vertical is always shown).
- Fixes a scrolling bug in `packages/compass-connections/src/components/connections.tsx` where the info under the connection form would not scroll. (I meant to break this into a separate commit and accidentally scooped up sry)

We have custom scrollbars mostly so that we can make it look nice on all platforms (windows looks rough without it).

Here's how they look and behave now:

https://user-images.githubusercontent.com/1791149/198158376-a65285ae-121c-4ff3-9d7e-b026987847b6.mp4

https://user-images.githubusercontent.com/1791149/198158754-4265aff2-f87b-4d2b-9dbf-602a8bd4ec0c.mp4

### CODE REVIEW

**Review by mcasimir — COMMENTED** [review]
🤑  This looks awesome! and super-clean implementation too!

I have a few questions/requests:
- what happens with modals/portals .. do we need to use it on the base components like Popover, etc .. or we can tell LG to always use some specific "root element" as a portal container?
- we may need to know the size of the scrollbar, so we can finally align the `WorkspaceContainer` content:

``` js
const {className, width} = useScrollbars();

const scrollbarMargin = useMemo(() => css({marginRight: -scrollbarWidth}), [scrollbarWidth]);
```

- Don't we need something that allows customizing the height so we could use this for the workspace tabs too?

Maybe  something like this:

``` js
const {className, width, height} = useScrollbars({width=10, height=10});
```

**Comment by Anemy** [comment]
Moving this to work in progress - we need to globally style these for all things using portals.

**Comment by mcasimir** [comment]
> Moving this to work in progress - we need to globally style these for all things using portals.

Let's consider cloud too

File: packages/compass-home/src/components/home.tsx [hunk 1]
```diff
@@ -273,14 +275,17 @@ function Home({
 
   if (isConnected) {
     return (
-      <div className="with-global-bootstrap-styles">
+      <div className={`with-global-bootstrap-styles ${scrollbarsClassName}`}>
```
Comment (gribnoysup) [inline packages/compass-home/src/components/home.tsx:278]: Why not `cx`? You already import stuff from compass components

File: packages/compass-shell/src/components/compass-shell/compass-shell.jsx [hunk 2]
```diff
@@ -21,10 +28,11 @@ const compassShellContainerStyles = css({
...
   overflow: 'auto',
   borderTop: `1px solid ${palette.gray.dark2}`,
-  '*::-webkit-scrollbar-thumb': {
-    background: rgba(palette.gray.light1, 0.5),
-  }
-});
+  // TODO(COMPASS-6206): After this package is on the new config
+  // we can start using a darkMode ThemeProvider in this component
+  // so we can do things like `useScrollbars()` without explicity
+  // using these scrollbar styles.
```
Comment (gribnoysup) [inline packages/compass-shell/src/components/compass-shell/compass-shell.jsx:34]: What's preventing you from using it on the current config? I am not sure how is this related 🤔

File: packages/compass-connections/src/components/connections.tsx [hunk 3]
```diff
@@ -114,7 +115,7 @@ function Connections({
           duplicateConnection={duplicateConnection}
         />
       </ResizableSidebar>
-      <div>
+      <>
```
Comment (gribnoysup) [inline packages/compass-connections/src/components/connections.tsx:118]: Is this fragment needed for something?

File: packages/compass-serverstats/src/components/index.tsx [hunk 4]
```diff
@@ -57,8 +63,10 @@ function PerformanceComponent() {
     };
   }, []);
 
+  const { className: scrollbarsClassName } = useScrollbars();
+
   return (
-    <section className="rt-perf">
+    <section className={`rt-perf ${scrollbarsClassName}`}>
```
Comment (gribnoysup) [inline packages/compass-serverstats/src/components/index.tsx:69]: Why are we explicitly applying these styles here? I thought this component is inside compass-home and shouldn't need these 🤔

File: packages/compass-shell/src/components/compass-shell/compass-shell.jsx (hunk 2 above)
Comment (Anemy) [reply packages/compass-shell/src/components/compass-shell/compass-shell.jsx:34]: I didn't want to convert it to a functional component without being able to also typescript it. Not totally a requirement, but figured it might make the swap a bit smoother.

**Comment by Anemy** [comment]
Close this pr as it's out of date and we'll likely do a slightly different approach.
